goalc/compiler/SymbolInfo.h

```diff
@@ -117,38 +153,54 @@ class SymbolInfo {
 class SymbolInfoMap {
  public:
   SymbolInfoMap() = default;
-  void add_global(const std::string& name, const goos::Object& defining_form) {
-    m_map[name]->push_back(SymbolInfo::make_global(name, defining_form));
+  void add_global(const std::string& name,
+                  const goos::Object& defining_form,
+                  const std::optional<SymbolInfo::Metadata> meta = {}) {
+    m_map[name]->push_back(SymbolInfo::make_global(name, defining_form, meta));
   }
 
   void add_fwd_dec(const std::string& name, const goos::Object& defining_form) {
     m_map[name]->push_back(SymbolInfo::make_fwd_declared_sym(name, defining_form));
   }
 
-  void add_function(const std::string& name, const goos::Object& defining_form) {
-    m_map[name]->push_back(SymbolInfo::make_function(name, defining_form));
+  // The m_symbol_types container stores TypeSpecs -- this does have argument information but not
+  // the names! The actual `LambdaVal` however does have all the types and names, so thats why its
```

Comment:
I think this comment is just out of date, but we should make sure to not store `LambdaVal*`s - these are supposed to be temporary just while a function is being compiled. Once the compiler moves on to the next file, they will be destroyed.